plugins/pulp_rpm/plugins/db/models.py

```diff
@@ -768,6 +768,12 @@ def merge_pkglists(self, other):
             # no collection with such name or no collection with such name and _pulp_repo_id
             else:
                 collections_to_add.append(new_collection)
+
+        # It is very important to call save() here to save recent modifications to the existing
+        # collections in the pkglist in the database before new collections will be added to
+        # the pkglist, because mongoengine does not allow to modify existing items in the list
+        # and add new items to the list at the same time.
+        self.save()
```

Comment:
Adding the additional `save()` would be good for the reason you mention in your comment.